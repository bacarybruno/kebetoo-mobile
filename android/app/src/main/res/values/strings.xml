<resources>
    <string name="app_name">Kebetoo</string>
</resources>
