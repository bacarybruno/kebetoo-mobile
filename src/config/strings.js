import LocalizedStrings from 'react-native-localization';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import 'dayjs/locale/en';
import 'dayjs/locale/fr';

import frStrings from '@assets/locales/fr';
import enStrings from '@assets/locales/en';

const strings = new LocalizedStrings({
  fr: frStrings,
  en: enStrings,
});

dayjs.locale(strings.getLanguage());
dayjs.extend(relativeTime);

export const updateAppLocale = (locale) => {
  dayjs.locale(locale);
  strings.setLanguage(locale);
};

export default strings;
