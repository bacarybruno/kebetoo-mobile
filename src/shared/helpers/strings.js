export const capitalize = (s) => {
  if (typeof s !== 'string') return ''
  const splitCharacters = '. '
  const parts = s.split(splitCharacters)
  return parts
    .map((s) => s.charAt(0).toUpperCase() + s.slice(1).toLowerCase())
    .join(splitCharacters)
}
