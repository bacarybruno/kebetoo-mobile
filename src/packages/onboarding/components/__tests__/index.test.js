import setupTest from 'Kebetoo/src/config/jest-setup'
import strings from 'Kebetoo/src/config/strings'

import OnboardingSlide from '../index'

const givenOnboardingSlide = setupTest(OnboardingSlide)()

it('renders OnboardingSlide', () => {
  const { wrapper } = givenOnboardingSlide({
    slideTitle: strings.onboarding.screen_one_title,
    slideDescription: strings.onboarding.screen_one_description,
    imageSrc: {
      uri: 'https://avatars1.githubusercontent.com/u/14147533',
    },
  })
  expect(wrapper.toJSON()).toMatchSnapshot()
})
