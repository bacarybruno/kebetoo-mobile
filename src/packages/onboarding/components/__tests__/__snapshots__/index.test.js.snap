// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Onboarding 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 16,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://avatars1.githubusercontent.com/u/14147533",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 600.3000000000001,
        "resizeMode": "contain",
      }
    }
  />
  <Text
    adjustsFontSizeToFit={true}
    style={
      Array [
        Object {
          "fontFamily": "Helvetica",
        },
        Object {
          "color": "rgba(0, 0, 0, 0.8)",
          "fontSize": 27,
          "fontWeight": "bold",
          "opacity": 1,
          "textDecorationLine": "none",
        },
        Object {
          "marginTop": 93.38000000000001,
          "maxHeight": 133.4,
        },
      ]
    }
  >
    Kebetoo is gonna change the way your communicate
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Helvetica",
        },
        Object {
          "color": "rgba(0, 0, 0, 0.8)",
          "fontSize": 16,
          "fontWeight": "normal",
          "opacity": 1,
          "textDecorationLine": "none",
        },
        Object {
          "marginTop": 46.690000000000005,
        },
      ]
    }
  >
    Just wait for it!
  </Text>
</View>
`;
